iso15118/secc/states/iso15118_2_states.py

```diff
@@ -1458,7 +1458,67 @@ async def process_message(
 
-        # TODO We should also do a more detailed check of the charging profile
+        # [V2G2-225] The SECC shall send the negative ResponseCode
+        # FAILED_ChargingProfileInvalid in
+        # the PowerDelivery response message if the EVCC sends a ChargingProfile which
+        # is not adhering to the PMax values of all PMaxScheduleEntry elements according
+        # to the chosen SAScheduleTuple element in the last ChargeParameterDiscoveryRes
+        # message sent by the SECC.
+        if power_delivery_req.charging_profile:
+            for schedule in self.comm_session.offered_schedules:
+                if (
+                    schedule.sa_schedule_tuple_id
+                    == power_delivery_req.sa_schedule_tuple_id
+                ):
+                    schedule_entries = schedule.p_max_schedule.schedule_entries
+                    # schedule_entries shall look like [PMaxScheduleEntry_0,
+                    # PMaxScheduleEntry_1, ...] The enumerate will pair an index with
+                    # each list entry: (0, PMaxScheduleEntry_0), (1,
+                    # PMaxScheduleEntry_1), ...
+                    for idx, sa_profile_entry in enumerate(schedule_entries):
+                        sa_profile_entry_start = sa_profile_entry.time_interval.start
+                        sa_entry_pmax = (
+                            sa_profile_entry.p_max.value
+                            * 10**sa_profile_entry.p_max.multiplier
+                        )
+
+                        for (
+                            ev_profile_entry
+                        ) in power_delivery_req.charging_profile.profile_entries:
+                            try:
+                                # By getting the next entry/slot, we can know when
+                                # the current entry ends
+                                next_sa_profile_entry = schedule_entries[idx + 1]
+                                sa_profile_entry_end = (
+                                    next_sa_profile_entry.time_interval.start
+                                )
+                            except IndexError:
+                                # As the index is out of rage, it signals this is
+                                # the final entry of the sa profile the last entry
+                                # must have a duration field, so we can calculate
+                                # the end of the period summing the relative start
+                                # of the entry with the duration of it
+                                sa_profile_entry_end = (
+                                    sa_profile_entry_start
+                                    + sa_profile_entry.time_interval.duration
+                                )
+                            if (
+                                sa_profile_entry_start
+                                <= ev_profile_entry.start
+                                <= sa_profile_entry_end
```

Comment:
this doesnt look readable...as it seems the problem is with the line break, we have three options:
1. either ignore the flake error by adding #noqa to the end and keeping the line as
`if sa_profile_entry_start <= ev_profile_entry.start <= sa_profile_entry_end:  #noqa`

2. or doing:
```python
if (
ev_profile_entry.start  >= sa_profile_entry_start and 
ev_profile_entry.start <= sa_profile_entry_end
):
```

3. This one is probably the best. As this part of the code is big and has too many indents (spaces), the best is to put the code in a private method within the class, like done here:
https://github.com/SwitchEV/iso15118/blob/0ee9d8ba853260d7ec38f8ff9b42339c3affa54b/iso15118/secc/states/iso15118_2_states.py#L1631

dont forget to prefix the "_" before the name of the private method

Reply:
Even i add the` #noqa` end of the line, black is still not happy. 
<img width="920" alt="image" src="https://user-images.githubusercontent.com/89934937/198578080-77b702ef-2b79-4683-9646-400f2c5fd432.png">

Reply:
fmt:on/off fixed this problem.